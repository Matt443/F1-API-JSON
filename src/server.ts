import { getDriverData } from "./scraper/driver-data";
import { getTeamsData } from "./scraper/team-data";
import { getDriverStandings } from "./scraper/driver-standings";
import { getConstructorStandings } from './scraper/constructors-standings';
import { getWorldChampions } from './scraper/world-champions'

export {
    getDriverData,
    getTeamsData,
    getDriverStandings,
    getConstructorStandings,
    getWorldChampions
}