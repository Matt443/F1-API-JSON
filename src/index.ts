import { getRaceResults } from "./server";

async function test() {
    await getRaceResults();
    console.log();
}
test();
