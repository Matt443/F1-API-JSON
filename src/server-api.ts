import express from "express";
import cors from "cors";
import helmet from "helmet";
import dotenv from "dotenv";

import router from "./endpoints-api/routes-api";
import { getFullRaceResults, getRaceResults } from "./server";

dotenv.config();

const app = express();

app.use(cors());
app.use(helmet());

app.use("/", router);

app.listen(process.env.PORT || 4005, () => {
    console.log(`server is running on ${process.env.PORT || 4005}`);
}).on("error", (error) => {
    throw new Error(error.message);
});
