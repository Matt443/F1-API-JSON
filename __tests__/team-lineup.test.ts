import { getTeamLineup } from "../src/server";

test("current teams data", async () => {
    expect(await getTeamLineup()).toMatchSnapshot();
});
