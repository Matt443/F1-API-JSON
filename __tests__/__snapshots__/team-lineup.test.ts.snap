// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`current teams data 1`] = `
[
  {
    "carImage": "https://www.formula1.com/content/dam/fom-website/teams/2022/red-bull-racing.png.transform/4col/image.png",
    "carLogo": "https://www.formula1.com/content/dam/fom-website/teams/2022/red-bull-racing-logo.png.transform/2col/image.png",
    "drivers": [
      "Max Verstappen",
      "Sergio Perez",
    ],
    "name": "Red Bull Racing",
    "points": 656,
    "rank": 1,
  },
  {
    "carImage": "https://www.formula1.com/content/dam/fom-website/teams/2022/ferrari.png.transform/4col/image.png",
    "carLogo": "https://www.formula1.com/content/dam/fom-website/teams/2022/ferrari-logo.png.transform/2col/image.png",
    "drivers": [
      "Carlos Sainz",
      "Charles Leclerc",
    ],
    "name": "Ferrari",
    "points": 469,
    "rank": 2,
  },
  {
    "carImage": "https://www.formula1.com/content/dam/fom-website/teams/2022/mercedes.png.transform/4col/image.png",
    "carLogo": "https://www.formula1.com/content/dam/fom-website/teams/2022/mercedes-logo.png.transform/2col/image.png",
    "drivers": [
      "George Russell",
      "Lewis Hamilton",
    ],
    "name": "Mercedes",
    "points": 416,
    "rank": 3,
  },
  {
    "carImage": "https://www.formula1.com/content/dam/fom-website/teams/2022/alpine.png.transform/4col/image.png",
    "carLogo": "https://www.formula1.com/content/dam/fom-website/teams/2022/alpine-logo.png.transform/2col/image.png",
    "drivers": [
      "Esteban Ocon",
      "Fernando Alonso",
    ],
    "name": "Alpine",
    "points": 144,
    "rank": 4,
  },
  {
    "carImage": "https://www.formula1.com/content/dam/fom-website/teams/2022/mclaren.png.transform/4col/image.png",
    "carLogo": "https://www.formula1.com/content/dam/fom-website/teams/2022/mclaren-logo.png.transform/2col/image.png",
    "drivers": [
      "Lando Norris",
      "Daniel Ricciardo",
    ],
    "name": "McLaren",
    "points": 138,
    "rank": 5,
  },
  {
    "carImage": "https://www.formula1.com/content/dam/fom-website/teams/2022/alfa-romeo.png.transform/4col/image.png",
    "carLogo": "https://www.formula1.com/content/dam/fom-website/teams/2022/alfa-romeo-logo.png.transform/2col/image.png",
    "drivers": [
      "Guanyu Zhou",
      "Valtteri Bottas",
    ],
    "name": "Alfa Romeo",
    "points": 52,
    "rank": 6,
  },
  {
    "carImage": "https://www.formula1.com/content/dam/fom-website/teams/2022/aston-martin.png.transform/4col/image.png",
    "carLogo": "https://www.formula1.com/content/dam/fom-website/teams/2022/aston-martin-logo.png.transform/2col/image.png",
    "drivers": [
      "Sebastian Vettel",
      "Lance Stroll",
    ],
    "name": "Aston Martin",
    "points": 51,
    "rank": 7,
  },
  {
    "carImage": "https://www.formula1.com/content/dam/fom-website/teams/2022/haas-f1-team.png.transform/4col/image.png",
    "carLogo": "https://www.formula1.com/content/dam/fom-website/teams/2022/haas-f1-team-logo.png.transform/2col/image.png",
    "drivers": [
      "Mick Schumacher",
      "Kevin Magnussen",
    ],
    "name": "Haas F1 Team",
    "points": 38,
    "rank": 8,
  },
  {
    "carImage": "https://www.formula1.com/content/dam/fom-website/teams/2022/alphatauri.png.transform/4col/image.png",
    "carLogo": "https://www.formula1.com/content/dam/fom-website/teams/2022/alphatauri-logo.png.transform/2col/image.png",
    "drivers": [
      "Pierre Gasly",
      "Yuki Tsunoda",
    ],
    "name": "AlphaTauri",
    "points": 36,
    "rank": 9,
  },
  {
    "carImage": "https://www.formula1.com/content/dam/fom-website/teams/2022/williams.png.transform/4col/image.png",
    "carLogo": "https://www.formula1.com/content/dam/fom-website/teams/2022/williams-logo.png.transform/2col/image.png",
    "drivers": [
      "Nicholas Latifi",
      "Alexander Albon",
    ],
    "name": "Williams",
    "points": 8,
    "rank": 10,
  },
]
`;
